image-rauc.c

```diff
@@ -161,6 +169,13 @@ static int rauc_parse(struct image *image, cfg_t *cfg)
 		list_add_tail(&part->list, &image->partitions);
 	}
 
+	for (i = 0; i < cfg_size(cfg, "intermediate"); i++) {
```

Comment:
The pkcs11 URL case is missing. Add partitions only for files and loop again in `rauc_generate()` do add the arguments for the URLs. You cannot use a partition for that: It will fail because the file is missing.

Hmm, @jluebbe, is the order of the `--intermediate` arguments relevant?

Reply:
RAUC keeps the order and forwards them to OpenSSL's CMS_sign as specified. I've not tested how different OpenSSL versions handle verification with multiple certs, so it's suggest to keep the same order here as well.

Reply:
AFAIK CMS_sign adds these certificates unconditionally to the binary and CMS_verify search of certificates does not care about the order.